### ISSUE TITLE
Caching with AnnotatedImage as output does not work on app reload

### ISSUE THREAD

**Issue by Blubbaa** [gradio-app/gradio#5521]
### Describe the bug

I am using the `gr.AnnotatedImage` component as an output in a simple interface. When I enable caching I noticed that the cached examples are only loaded and displayed correctly in the first startup (that also creates the cache). Any subsequent app starts will not display any output but also no error message. I have condensed my code to a minimal working example below (using some fake annotation from the docs).

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```python
import random

import gradio as gr
import numpy as np

MAX_WIDTH = 640
MAX_HEIGHT = 480

SECTION_LABELS = [
    "apple",
    "banana",
    "carrot",
    "donut",
    "eggplant",
    "fish",
    "grapes",
    "hamburger",
    "ice cream",
    "juice",
]


def section(img, num_boxes, num_segments):
    sections = []
    for a in range(num_boxes):
        x = random.randint(0, img.shape[1])
        y = random.randint(0, img.shape[0])
        w = random.randint(0, img.shape[1] - x)
        h = random.randint(0, img.shape[0] - y)
        sections.append(((x, y, x + w, y + h), SECTION_LABELS[a]))
    for b in range(num_segments):
        x = random.randint(0, img.shape[1])
        y = random.randint(0, img.shape[0])
        r = random.randint(0, min(x, y, img.shape[1] - x, img.shape[0] - y))
        mask = np.zeros(img.shape[:2])
        for i in range(img.shape[0]):
            for j in range(img.shape[1]):
                dist_square = (i - y) ** 2 + (j - x) ** 2
                if dist_square < r**2:
                    mask[i, j] = round((r**2 - dist_square) / r**2 * 4) / 4
        sections.append((mask, SECTION_LABELS[b + num_boxes]))
    return (img, sections)


# Create interface function
def create_annotation(input_image, other_args=1):
    # Generate random annotations
    output = section(input_image, num_boxes=3, num_segments=2)

    # return output tuple
    return output


if __name__ == "__main__":
    # some example images
    examples = [
        "image/0000155DA.jpg",
        "image/0000155DF.jpg",
    ]

    # Create a simple Gradio interface
    simple_demo = gr.Interface(
        fn=create_annotation,
        inputs=gr.Image(width=MAX_WIDTH, height=MAX_HEIGHT),
        outputs=gr.AnnotatedImage(width=MAX_WIDTH, height=MAX_HEIGHT),
        analytics_enabled=False,
        examples=examples,
        cache_examples=True,
    )
    simple_demo.launch(inbrowser=True)

```

### System Info

```shell
Gradio Environment Information:
------------------------------ 
Operating System: Windows
gradio version: 3.43.2
gradio_client version: 0.5.0

------------------------------------------------
gradio dependencies in your environment:        

aiofiles: 23.1.0
altair: 4.2.2
fastapi: 0.95.1
ffmpy: 0.3.0
gradio-client==0.5.0 is not installed.
httpx: 0.24.0
huggingface-hub: 0.16.4
importlib-resources: 5.12.0
jinja2: 3.1.2
markupsafe: 2.1.2
matplotlib: 3.7.1
numpy: 1.24.4
orjson: 3.8.10
packaging: 23.1
pandas: 1.5.3
pillow: 9.5.0
pydantic: 1.10.7
pydub: 0.25.1
python-multipart: 0.0.6
pyyaml: 6.0
requests: 2.28.2
semantic-version: 2.10.0
typing-extensions: 4.5.0
uvicorn: 0.21.1
websockets: 11.0.2
authlib; extra == 'oauth' is not installed.
itsdangerous; extra == 'oauth' is not installed.


gradio_client dependencies in your environment:

fsspec: 2023.4.0
httpx: 0.24.0
huggingface-hub: 0.16.4
packaging: 23.1
requests: 2.28.2
typing-extensions: 4.5.0
websockets: 11.0.2
```

### Severity

I can work around it

**Comment by abidlabs** [comment]
Hi @Blubbaa I can't reproduce this locally or on Colab: https://colab.research.google.com/drive/1SY861i3eFeTVKWE7C8w39cBuHF8vMlXc?usp=sharing

Can you see if this is still an issue in the latest version of `gradio`?

Also can you clarify what you mean by: 

> Any subsequent app starts

**Comment by Blubbaa** [comment]
I tried with the latest version, but the issue is still there. With subsequent app starts I mean:
1. launch the interface the first time, cache gets built, everything works
2. close the server
3. launch the demo a second time, it says its reusing the cache but clicking on examples has no effect

Indeed I am able to reproduce also in Colab with your notebook:

1. Select runtime > run all (everything fine)
2. Restart runtime
3. Run demo > clicking on examples has no effect

**Comment by abidlabs** [comment]
Ok yup can reproduce now! Will take a look

**Comment by abidlabs** [comment]
Confirmed that this is closed by #5636 -- thanks @freddyaboulton!
